document/core/appendix/properties.rst

```diff
@@ -128,23 +313,26 @@ Module instances are classified by *module contexts*, which are regular :ref:`co
      \qquad
      (S \vdashdatainst \datainst \ok)^\ast
      \\
+     (S \vdashstructinst \structinst \ok)^\ast
+     \qquad
+     (S \vdasharrayinst \arrayinst \ok)^\ast
+     \\
      S = \{
        \begin{array}[t]{@{}l@{}}
        \SFUNCS~\funcinst^\ast,
-       \SGLOBALS~\globalinst^\ast, \\
+       \SGLOBALS~\globalinst^\ast,
        \STABLES~\tableinst^\ast,
-       \SMEMS~\meminst^\ast,
+       \SMEMS~\meminst^\ast, \\
        \SELEMS~\eleminst^\ast,
-       \SDATAS~\datainst^\ast \}
+       \SDATAS~\datainst^\ast,
+       \SSTRUCTS~\structinst^\ast,
+       \SARRAYS~\arrayinst^\ast \}
        \end{array}
      \end{array}
    }{
      \vdashstore S \ok
    }
 
-.. note::
-   The validity condition on type instances ensures the absence of cyclic types.
```

Comment:
This note has been outdated for a while, I just missed it. We no longer have type instances. And since we no longer have indirections through a type store, types are closed syntactic terms and hence non-cyclic by construction.